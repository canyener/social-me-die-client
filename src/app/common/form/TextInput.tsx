import { FieldRenderProps } from "react-final-form"
import { Form, FormFieldProps, Label } from "semantic-ui-react"

interface IProps
  extends FieldRenderProps<string, HTMLInputElement>,
  FormFieldProps { }

const TextInput: React.FC<IProps> = ({
  input,
  width,
  type,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <input type={type} {...input} placeholder={placeholder} />
      {
        touched && error && (
          <Label basic color='red'>
            {error}
          </Label>
        )
      }
    </Form.Field>
  )
}

export default TextInput 