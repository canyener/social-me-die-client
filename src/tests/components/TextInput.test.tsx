import { shallow } from 'enzyme'
import TextInput from '../../app/common/form/TextInput'

test('Should render TextInput correctly without any given errors', () => {

  const wrapper = shallow(<TextInput
    type='text'
    width='11'
    placeholder='Test Input PlaceHolder'
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test input name',
      value: 'Test input value'
    }}
    meta={{
      touched: false,
      error: undefined
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})

test('Should set Form.Field error to true and render Label with given error message', () => {
  const wrapper = shallow(<TextInput
    type='text'
    width='11'
    placeholder='Test Input PlaceHolder'
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test input name',
      value: 'Test input value'
    }}
    meta={{
      touched: true,
      error: 'test error message'
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})