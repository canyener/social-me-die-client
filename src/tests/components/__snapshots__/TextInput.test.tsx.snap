// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TextInput correctly without any given errors 1`] = `
<FormField
  error={false}
  width="11"
>
  <input
    name="Test input name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Test Input PlaceHolder"
    value="Test input value"
  />
</FormField>
`;

exports[`Should set Form.Field error to true and render Label with given error message 1`] = `
<FormField
  error={true}
  width="11"
>
  <input
    name="Test input name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Test Input PlaceHolder"
    value="Test input value"
  />
  <Label
    basic={true}
    color="red"
  >
    test error message
  </Label>
</FormField>
`;
